DirectX11/HackerDXGI.cpp

```diff
@@ -167,6 +167,104 @@ HackerDXGISwapChain::HackerDXGISwapChain(IDXGISwapChain *pSwapChain, HackerDevic
 	}
 }
 
+HackerDXGISwapChain::HackerDXGISwapChain(IDXGISwapChain *pSwapChain, HackerDevice *pDevice, HackerContext *pContext, const DXGI_SWAP_CHAIN_DESC* fake_swap_chain_desc, UINT new_width, UINT new_height)
+	: HackerDXGIDeviceSubObject(pSwapChain), mFakeBackBuffer(nullptr), mFakeSwapChain(nullptr), mWidth(0), mHeight(0)
+{
+	mOrigSwapChain = pSwapChain;
+
+	if (pContext == NULL)
+	{
+		pDevice->GetImmediateContext(reinterpret_cast<ID3D11DeviceContext**>(&pContext));
+	}
+
+	mHackerDevice = pDevice;
+	mHackerContext = pContext;
+	pDevice->SetHackerSwapChain(this);
+
+	// TODO: multisampled swap chain
+	// TODO: swap chain with more than one back buffer
+	D3D11_TEXTURE2D_DESC fake_buffer_desc;
+	std::memset(&fake_buffer_desc, 0, sizeof(D3D11_TEXTURE2D_DESC));
+	fake_buffer_desc.ArraySize = 1;
+	fake_buffer_desc.MipLevels = 1;
+	fake_buffer_desc.BindFlags = D3D11_BIND_RENDER_TARGET | D3D11_BIND_SHADER_RESOURCE;
+	fake_buffer_desc.Usage = D3D11_USAGE_DEFAULT;
+	fake_buffer_desc.SampleDesc.Count = 1;
+	fake_buffer_desc.Format = fake_swap_chain_desc->BufferDesc.Format;
+	fake_buffer_desc.MiscFlags = 0;
+	fake_buffer_desc.Width = fake_swap_chain_desc->BufferDesc.Width;
+	fake_buffer_desc.Height = fake_swap_chain_desc->BufferDesc.Height;
+	fake_buffer_desc.CPUAccessFlags = 0;
+
+	HRESULT res = pDevice->CreateTexture2D(&fake_buffer_desc, nullptr, &mFakeBackBuffer);
+
+	mWidth = new_width;
+	mHeight = new_height;
+
+	if (res != S_OK)
+	{
+		LogInfo(" *** Failed to create faked swap chain buffer! \n");
+		mFakeBackBuffer = nullptr;
+		throw;
```

Comment:
Hmmm... Is 'throw' the only option here?  That will just crash silently in a lot of games, with no way for them to know what happened.

If the swap chain texture fails, wouldn't it make more sense to fall back to normal behavior instead?  Probably disable the upscaling and just return?

Our default for configuration errors is to make a double-boop sound (found in Utils.)  But keep going.  That way the user knows something is wrong, but it's not a fatal error.